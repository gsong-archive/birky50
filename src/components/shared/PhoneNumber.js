//@flow strict
import React from "react";

import { Link } from "../../styles/components";
import { Phone } from "./EmojiLabels";

type Props = { number: string };

export default ({ number }: Props) => (
  <div>
    <Phone label={<Link href={`tel:${number}`}>{number}</Link>} />
  </div>
);
