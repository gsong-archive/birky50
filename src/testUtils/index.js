import * as faker from "faker";

const address = () =>
  [
    faker.address.streetAddress(),
    faker.address.city(),
    `${faker.address.stateAbbr()} ${faker.address.zipCode()}`,
  ].join(", ");

const card = () => ({
  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
  url: faker.internet.url(),
  price: faker.random.word(),
  address: address(),
  addressUrl: faker.internet.url(),
  phone: faker.phone.phoneNumber(),
  imgUrl: faker.image.imageUrl(),
});

const generate = {
  address,
  card,
  phoneNumber: faker.phone.phoneNumber,
  url: faker.internet.url,
  words: faker.random.words,
};

export { generate };
