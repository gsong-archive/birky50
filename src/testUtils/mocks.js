import * as cssFeatures from "../styles/cssFeatures";
import * as styleUtils from "../styles/utils";

export const mockGetBaseFontSize = {
  mock: () => (styleUtils.getBaseFontSize = jest.fn(() => 16)),
  restore: () => styleUtils.getBaseFontSize.mockRestore(),
};

export const mockSupportsGrid = bool => (cssFeatures.supportsGrid = bool);

export const mockSupportsObjectFit = bool =>
  (cssFeatures.supportsObjectFit = bool);
