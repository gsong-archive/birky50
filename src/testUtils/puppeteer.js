import "../env";

const DEFAULT_APP_URL = "http://localhost:3000";
const DEFAULT_JEST_TIMEOUT = 5000;
const DEFAULT_SCREENSHOTS_FOLDER = "screenshots/puppeteer";
const DEFAULT_SCROLL_CHECK_DELAY = 100;

const appUrl =
  process.env.APP_URL === undefined ? DEFAULT_APP_URL : process.env.APP_URL;

const jestTimeOut =
  parseInt(process.env.JEST_TIMEOUT, 10) || DEFAULT_JEST_TIMEOUT;

const screenshotsFolder =
  process.env.PUPPETEER_SCREENSHOTS_FOLDER === undefined
    ? DEFAULT_SCREENSHOTS_FOLDER
    : process.env.PUPPETEER_SCREENSHOTS_FOLDER;

const scrollCheckDelay =
  process.env.PUPPETEER_SCROLL_CHECK_DELAY || DEFAULT_SCROLL_CHECK_DELAY;

export const settings = { appUrl, jestTimeOut, scrollCheckDelay };

export const screenshot = async (imageName, page, options = {}) => {
  await page.screenshot({
    path: `${screenshotsFolder}/${imageName}.png`,
    ...options,
  });
};

const desktops = {
  "macbook-13": { width: 1280, height: 800 },
  "macbook-15": { width: 1440, height: 900 },
};

const mobiles = {
  "ipad-portrait": { width: 768, height: 1024 },
  "iphone-se-portrait": { width: 320, height: 568 },
  "iphone-8-portrait": { width: 375, height: 667 },
  "iphone-8-plus-portrait": { width: 540, height: 960 },
  "iphone-x-portrait": { width: 375, height: 812 },
};

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device.replace("portrait", "landscape")] = {
    width: viewport.height,
    height: viewport.width,
    isLandscape: true,
  };
});

Object.entries(mobiles).forEach(([device, viewport]) => {
  mobiles[device] = { ...viewport, isMobile: true, hasTouch: true };
});

export const devices = Object.assign({}, desktops, mobiles);
