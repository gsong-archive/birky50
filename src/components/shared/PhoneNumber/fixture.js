//@flow
import { createFixture } from "react-cosmos-flow/fixture";

import { generate } from "testUtils";

import PhoneNumber from "./index";

export default createFixture({
  component: PhoneNumber,
  name: "PhoneNumber",
  props: { number: generate.phoneNumber() },
});
