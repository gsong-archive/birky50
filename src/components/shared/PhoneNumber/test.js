//@flow strict
import React from "react";

import { render } from "react-testing-library";

import { generate } from "testUtils";

import PhoneNumber from "./index";

test("User can see phone number and link", () => {
  const props = { number: generate.phoneNumber() };
  const { container } = render(<PhoneNumber {...props} />);
  const link = container.querySelector("a");

  expect(container).toHaveTextContent(props.number);
  expect(link.href).toBe(`tel:${props.number}`);
});

test("PhoneNumber is rendered", () => {
  const props = { number: "503-123-4567" };
  const { container } = render(<PhoneNumber {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
