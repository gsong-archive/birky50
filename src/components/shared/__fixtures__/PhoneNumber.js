import { generate } from "testUtils";

import PhoneNumber from "../PhoneNumber";

export default {
  component: PhoneNumber,
  props: { number: generate.phoneNumber() },
};
