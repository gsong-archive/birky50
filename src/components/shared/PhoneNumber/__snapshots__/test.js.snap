// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumber is rendered 1`] = `
.emotion-3 {
  display: inline-block;
  margin-left: 1.4em;
  text-indent: -1.4em;
}

.emotion-0 {
  font-family: "Segoe UI Emoji";
}

.emotion-1 {
  color: maroon;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div>
  <span
    class="emotion-3"
  >
    <span
      class="emotion-0"
      role="img"
      aria-label="phone receiver"
    >
      ðŸ“ž
    </span>
    <a
      href="tel:503-123-4567"
      class="emotion-1 emotion-2"
    >
      503-123-4567
    </a>
  </span>
</div>
`;
